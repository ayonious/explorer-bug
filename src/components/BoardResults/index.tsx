import BugReportIcon from '@mui/icons-material/BugReport';
import FilterVintageIcon from '@mui/icons-material/FilterVintage';
import React from 'react';

import {
  COL_LIMIT,
  robotPosToFlightStyle,
  RobotState,
  ROW_LIMIT,
} from '../../helpers/DirectionManager';
import { Cell, Grid, Row } from './styles';

const inputToPlayerIcon = (
  robotState: RobotState,
  cellState: number,
  rowId: number,
  colId: number
) => {
  const isLocationForPlayer =
    robotState.col === colId && robotState.row === rowId;
  const isLocationForFood = !isLocationForPlayer && !!cellState;
  const isBlank = !isLocationForPlayer && !isLocationForFood;

  return (
    <div
      style={
        isLocationForPlayer ? robotPosToFlightStyle(robotState) : undefined
      }
    >
      {isLocationForPlayer && <BugReportIcon />}
      {isLocationForFood && <FilterVintageIcon />}
      {isBlank && ''}
    </div>
  );
};

const CreateCell = (props: {
  robotState: RobotState;
  cellState: number;
  rowId: number;
  colId: number;
}) => {
  const { robotState, rowId, colId, cellState } = props;
  return (
    <Cell data-testid={`cell-${rowId}-${colId}`}>
      {inputToPlayerIcon(robotState, cellState, rowId, colId)}
    </Cell>
  );
};

const CreateRow = (props: {
  robotState: RobotState;
  rowGrid: number[];
  colCount: number;
  rowId: number;
}) => {
  const { robotState, colCount, rowId, rowGrid } = props;
  const columnList = Array(colCount)
    .fill(0)
    .map((e, colId) => (
      <CreateCell
        key={`colId:${colId}-rowId:${rowId}`}
        robotState={robotState}
        cellState={rowGrid[colId]}
        rowId={rowId}
        colId={colId}
      />
    ));
  return <Row>{columnList}</Row>;
};

const CreateRows = (props: {
  robotState: RobotState;
  gridState: number[][];
  rowCount: number;
}) => {
  const { robotState, rowCount, gridState } = props;
  const rowList = Array(rowCount)
    .fill(0)
    .map((e, rowId) => (
      <CreateRow
        key={`rowId-${rowId}`}
        robotState={robotState}
        rowGrid={gridState[rowId]}
        rowId={rowId}
        colCount={COL_LIMIT}
      />
    ));
  return <div>{rowList}</div>;
};
interface Props {
  robotState: RobotState;
  gridState: number[][];
}

const BoardResults = (props: Props) => {
  return (
    <Grid data-testid="game-board">
      <CreateRows
        robotState={props.robotState}
        gridState={props.gridState}
        rowCount={ROW_LIMIT}
      />
    </Grid>
  );
};

export default BoardResults;
