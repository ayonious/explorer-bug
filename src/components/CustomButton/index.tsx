import Button from '@mui/material/Button';
import * as React from 'react';

import { CustomButtonDiv } from './styles';

const CustomButton = (props: any) => {
  return (
    <CustomButtonDiv>
      <Button variant="contained" color="primary" onClick={props.onClick}>
        {props.title}
      </Button>
    </CustomButtonDiv>
  );
};

export default CustomButton;
