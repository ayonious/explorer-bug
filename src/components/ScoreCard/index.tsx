import * as React from 'react';

import { CustomDiv } from './styles';

const ScoreCard = ({ score }: { score: number }) => {
  return (
    <CustomDiv>
      <h1> Score: {score} </h1>
    </CustomDiv>
  );
};

export default ScoreCard;
