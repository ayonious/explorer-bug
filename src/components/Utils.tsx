import Button from '@material-ui/core/Button';
import * as React from 'react';

import { CustomButtonDiv, CustomDiv } from '../styledcomponents/styles';

export function CustomButton(props: any) {
  return (
    <CustomButtonDiv>
      <Button variant="contained" color="primary" onClick={props.onClick}>
        {props.title}
      </Button>
    </CustomButtonDiv>
  );
}

export function Greeting() {
  return (
    <CustomDiv>
      <h1> Move the bug to eat all food! </h1>
    </CustomDiv>
  );
}

export function ScoreCard({ score }: { score: number }) {
  return (
    <CustomDiv>
      <h1> Score: {score} </h1>
    </CustomDiv>
  );
}
