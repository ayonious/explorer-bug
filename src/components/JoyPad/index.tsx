import * as React from 'react';

import { Row } from './styles';
import CustomButton from '../CustomButton';

interface Props {
  handleLeft: any;
  handleRight: any;
  handleGo: any;
}

const JoyPad = (props: Props) => {
  return (
    <div>
      <Row>
        <CustomButton onClick={props.handleGo} title={'Go'} />
      </Row>
      <Row>
        <CustomButton onClick={props.handleLeft} title={'Left'} />
        <CustomButton onClick={props.handleRight} title={'Right'} />
      </Row>
    </div>
  );
};

export default JoyPad;
