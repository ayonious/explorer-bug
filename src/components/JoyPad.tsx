import * as React from 'react';
import { Row } from '../styledcomponents/styles';
import { CustomButton } from './Utils';

interface Props {
  handleLeft: any;
  handleRight: any;
  handleGo: any;
}

interface State {}

export default class JoyPad extends React.Component<Props, State> {
  render() {
    return (
      <div>
        <Row>
          <CustomButton onClick={this.props.handleGo} title={'Go'} />
        </Row>
        <Row>
          <CustomButton onClick={this.props.handleLeft} title={'Left'} />
          <CustomButton onClick={this.props.handleRight} title={'Right'} />
        </Row>
      </div>
    );
  }
}
