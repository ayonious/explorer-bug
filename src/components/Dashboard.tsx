import * as React from 'react';

import {
  COL_LIMIT,
  getnewGrid,
  getnewScore,
  moveForward,
  RobotState,
  ROW_LIMIT,
  turnLeft,
  turnRight,
} from '../helpers/DirectionManager';
import { MainDashboard } from '../styledcomponents/styles';
import BoardReults from './BoardResults';
import JoyPad from './JoyPad';
import { Greeting, ScoreCard } from './Utils';

interface Props {}

interface State {
  robotState: RobotState;
  liveInputScript: string;
  score: number;
  gridState: number[][];
}

const randomTossProbabilibty15 = () => {
  return Math.floor(Math.random() * 100 + 1) < 15 ? 1 : 0;
};

// generate a grid, if 1 then there is a food otherwise no food
const genFoodLocation = (rowNums: number, colNums: number) => {
  const ret: number[][] = [];
  for (let i = 0; i < rowNums; i++) {
    const rowArr: number[] = [];
    for (let j = 0; j < colNums; j++) {
      rowArr.push(randomTossProbabilibty15());
    }
    ret.push(rowArr);
  }
  return ret;
};

export default class Dashboard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.changedInput = this.changedInput.bind(this);
    this.handleGo = this.handleGo.bind(this);
    this.handleLeft = this.handleLeft.bind(this);
    this.handleRight = this.handleRight.bind(this);
    this.state = {
      liveInputScript: '',
      robotState: {
        row: 0,
        col: 0,
        direction: 0,
      },
      score: 0,
      gridState: genFoodLocation(ROW_LIMIT, COL_LIMIT),
    };
  }

  changedInput(event: any) {
    this.setState({ liveInputScript: event.target.value });
  }

  handleGo() {
    const newLocation: RobotState = moveForward(this.state.robotState);
    this.setState({
      robotState: newLocation,
      score: getnewScore(newLocation, this.state.score, this.state.gridState),
      gridState: getnewGrid(newLocation, this.state.gridState),
    });
  }

  handleLeft() {
    this.setState({
      robotState: turnLeft(this.state.robotState),
    });
  }

  handleRight() {
    this.setState({
      robotState: turnRight(this.state.robotState),
    });
  }

  render() {
    return (
      <MainDashboard>
        <div>
          <Greeting />
          <ScoreCard score={this.state.score} />
        </div>

        <JoyPad
          handleGo={this.handleGo}
          handleLeft={this.handleLeft}
          handleRight={this.handleRight}
        />

        <div>
          {' '}
          <br />{' '}
        </div>

        <BoardReults
          robotState={this.state.robotState}
          gridState={this.state.gridState}
        />

        <div>
          {' '}
          <br />{' '}
        </div>
      </MainDashboard>
    );
  }
}
