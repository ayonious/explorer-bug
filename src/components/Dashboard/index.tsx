import * as React from 'react';
import GithubCorner from 'react-github-corner';

import {
  COL_LIMIT,
  getNewGrid,
  getNewScore,
  moveForward,
  RobotState,
  ROW_LIMIT,
  turnLeft,
  turnRight,
} from '../../helpers/DirectionManager';
import BoardResults from '../BoardResults';
import Greeting from '../Greeting';
import JoyPad from '../JoyPad';
import ScoreCard from '../ScoreCard';
import { MainDashboard } from './styles';
import GithubCornerPart from './GitCorner';

interface Props {}

interface State {
  robotState: RobotState;
  liveInputScript: string;
  score: number;
  gridState: number[][];
}

const randomTossProbabilibty15 = () => {
  return Math.floor(Math.random() * 100 + 1) < 15 ? 1 : 0;
};

// generate a grid, if 1 then there is a food otherwise no food
const genFoodLocation = (rowNums: number, colNums: number): number[][] => {
  return [...Array(rowNums)].map(() => {
    return [...Array(colNums)].map(() => randomTossProbabilibty15());
  });
};

export default class Dashboard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.changedInput = this.changedInput.bind(this);
    this.handleGo = this.handleGo.bind(this);
    this.handleLeft = this.handleLeft.bind(this);
    this.handleRight = this.handleRight.bind(this);
    this.state = {
      liveInputScript: '',
      robotState: {
        row: 0,
        col: 0,
        direction: 0,
      },
      score: 0,
      gridState: genFoodLocation(ROW_LIMIT, COL_LIMIT),
    };
  }

  changedInput(event: any) {
    this.setState({ liveInputScript: event.target.value });
  }

  handleGo() {
    const newLocation: RobotState = moveForward(this.state.robotState);
    this.setState({
      robotState: newLocation,
      score: getNewScore(newLocation, this.state.score, this.state.gridState),
      gridState: getNewGrid(newLocation, this.state.gridState),
    });
  }

  handleLeft() {
    this.setState({
      robotState: turnLeft(this.state.robotState),
    });
  }

  handleRight() {
    this.setState({
      robotState: turnRight(this.state.robotState),
    });
  }

  render() {
    return (
      <MainDashboard>
        <GithubCornerPart />
        <div>
          <Greeting />
          <ScoreCard score={this.state.score} />
        </div>

        <JoyPad
          handleGo={this.handleGo}
          handleLeft={this.handleLeft}
          handleRight={this.handleRight}
        />

        <div>
          {' '}
          <br />{' '}
        </div>

        <BoardResults
          robotState={this.state.robotState}
          gridState={this.state.gridState}
        />

        <div>
          {' '}
          <br />{' '}
        </div>
      </MainDashboard>
    );
  }
}
