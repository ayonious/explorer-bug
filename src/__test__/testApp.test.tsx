import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ScoreCard } from '../components/Utils';

it('CountryResults renders', () => {
  const tree = renderer.create(<ScoreCard score={10} />);
  expect(tree).toMatchSnapshot();
});
