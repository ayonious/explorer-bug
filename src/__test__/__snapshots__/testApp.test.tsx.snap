// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryResults renders 1`] = `
<div
  className="sc-AykKD lfrAza"
>
  <h1>
     Score: 
    10
     
  </h1>
</div>
`;
